### ISSUE TITLE
`NoneType` dereference in `generate_psk`

### ISSUE THREAD

**Issue by vemek** [ggravlingen/pytradfri#128]
### Where are you using pytradfri (eg stand-alone, Home Assistant etc)
Latest Home Assistant, fresh install

### Version of pytradfri
4.1.0

### Expected behaviour
pytradfri creates a PSK for the hub

### Actual behaviour
We get a value of `None` for `security_key`:

```
2017-12-18 12:54:31 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/home/vemek/.pyenv/versions/3.5.3/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/home/vemek/.pyenv/versions/3.5.3/envs/hass/lib/python3.5/site-packages/homeassistant/core.py", line 1031, in _event_to_service_call
    yield from service_handler.func(service_call)
  File "/home/vemek/.pyenv/versions/3.5.3/envs/hass/lib/python3.5/site-packages/homeassistant/components/configurator.py", line 224, in async_handle_service_call
    call.data.get(ATTR_FIELDS, {}))
  File "/home/vemek/.pyenv/versions/3.5.3/lib/python3.5/asyncio/futures.py", line 380, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/home/vemek/.pyenv/versions/3.5.3/lib/python3.5/asyncio/tasks.py", line 304, in _wakeup
    future.result()
  File "/home/vemek/.pyenv/versions/3.5.3/lib/python3.5/asyncio/futures.py", line 293, in result
    raise self._exception
  File "/home/vemek/.pyenv/versions/3.5.3/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/home/vemek/.pyenv/versions/3.5.3/envs/hass/lib/python3.5/site-packages/homeassistant/components/tradfri.py", line 70, in configuration_callback
    key = yield from api_factory.generate_psk(security_code)
  File "/home/vemek/.pyenv/versions/3.5.3/envs/hass/lib/python3.5/site-packages/pytradfri/api/aiocoap_api.py", line 193, in generate_psk
    PatchedDTLSSecurityStore.KEY = security_key.encode('utf-8')
AttributeError: 'NoneType' object has no attribute 'encode'
```

It's worth noting that `example_async` works as expected and detects the hub and can read the state of the lights. I'm not sure why this is not working under Home Assistant, but I'm more than happy to help debug if you could kindly point me in the right direction? I'm going to take a look at the Home Assistant component source and see if I can create minimal reproduction steps using just pytradfri. Thanks for all your work on this!

**Comment by Lakitna** [comment]
This is a bug that has been fixed in 5.0.0

See the old issue #96 for details.

**Comment by vemek** [comment]
Thanks for the context @Lakitna! It looks like that change was merged into HA 0.58, but I'm on 0.60 and seeing the issue surface. I wonder if this is a related but separate problem 🤔

**Comment by vemek** [comment]
After some debugging, it looks like the issue is that HA is passing `None` as the `security_code`. I think this is a HA issue so, although perhaps the parameter should be validated by the method in pytradfri with a clear error message. I'll PR that if I can. Thanks again.
